compiler/luci/lang/include/luci/IR/ProfilingData.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <loco.h>
+
+#include <set>
+
+namespace luci
+{
+
+/**
+ * @brief Set annotation for Profiling data
+ */
+class ProfilingData final : public loco::NodeAnnotation
+{
+public:
+  ProfilingData()
+  {
+    // DO NOTHING
+  }
+
+  ProfilingData(int32_t id)
+  {
+    _node_id = id;
+    _origins.insert(id);
+  }
+
+  ProfilingData(uint32_t id)
+  {
+    _node_id = static_cast<int32_t>(id);
+    _origins.insert(static_cast<int32_t>(id));
+  }
+
+public:
+  int32_t node_id(void) const { return _node_id; }
+  void node_id(int32_t id) { _node_id = id; }
```

Comment:
Is `node id` same concept with 'operator index'?

Reply:
Exactly same for now :)
Sorry for poor naming :(
Do you want to rename like `op_index()`?

Reply:
Not necessarily, it maybe good to have independent name considering future modification.

But I hope comments to be added that this is operator index as of now on main branch :)

Reply:
Also comment on meaning of negative node id looks required if `int32_t` used.

Alternative design, we may use `uint32_t` for this. I guess this node would have valid unsigned value in all cases?

Reply:
I used `int32_t` to represent undefined status because operator index is started from 0.
Not special reason :)

I think we can use `uint32_t` and delete default constructor since node id will not be changed once it is allocated :)